### ISSUE TITLE
Can't upload big file

### ISSUE THREAD

**Issue by tamdao** [3846masa/upload-gphotos#331]
I have tried to implement feature upload big file. I can upload the big file with my code, but I don't see the file on google photo after upload success. Can you help me take a look my code?

```js
'use strict';
const Promise = require('bluebird');
const fs = require('fs');

const url = 'https://photos.google.com';
function apiInit(args, videoSize) {
  const options = {
    method: 'POST',
    url: `${url}/_/upload/uploadmedia/interactive?authuser=1`,
    headers: {
      'x-goog-upload-command': 'start',
      'x-goog-upload-header-content-length': videoSize,
      'x-goog-upload-protocol': 'resumable'
    }
  };
  return args.axios(options)
    .then(res => {
      return {
        granularity: +res.headers['x-goog-upload-chunk-granularity'],
        url: res.headers['x-goog-upload-url'],
        uploadid: res.headers['x-guploader-uploadid'],
        contentLength: videoSize,
      };
    });
}

function apiFinish(args, filePath, uploadInfo) {
  const options = {
    method: 'POST',
    url: `${url}/_/upload/uploadmedia/interactive?authuser=1&upload_id=${uploadInfo.uploadid}&upload_protocol=resumable`,
    headers: {
      'x-goog-upload-command': 'upload, finalize',
      'x-goog-upload-offset': 0,
    },
    data: fs.createReadStream(filePath, {
      encoding: null,
      start: 0,
      end: uploadInfo.contentLength,
    }),
  };
  return args.axios(options)
    .then(res => {
      return res;
    });
}

function uploadChunk(args, filePath, uploadUrl, chunk) {
  const options = {
    method: 'POST',
    url: uploadUrl,
    headers: {
      'x-goog-upload-command': 'upload',
      'x-goog-upload-offset': chunk.bstart
    },
    body: fs.createReadStream(filePath, {
      encoding: null,
      start: chunk.bstart,
      end: chunk.bend + 1
    }),
  };

  return args.axios(options);
}

function getChunks(filePath, granularity, start) {
  const allsize = fs.statSync(filePath).size;
  const sep = allsize < (granularity) ? allsize : (granularity) - 1;
  const ar = [];
  for (let i = start; i < allsize; i += sep) {
    const bstart = i;
    const bend = i + sep - 1 < allsize ? i + sep - 1 : allsize - 1;
    ar.push({
      bstart: bstart,
      bend: bend,
    });
  }
  return ar;
}

async function initUploadChain(args, filePath, uploadInfo) {
  const chunks = getChunks(filePath, uploadInfo.granularity, 0);
  let i = 0;
  while (i < chunks.length) {
    console.info('Uploading chunk ' + i + ' start: ' + chunks[i].bstart);
    await uploadChunk(args, filePath, uploadInfo.url, chunks[i]);
    i++;
  }
  return apiFinish(args, filePath, uploadInfo);
}

function getFileSize(filePath) {
  return new Promise((resolve, reject) => {
    fs.stat(filePath, function(err, stats) {
      if(err) {
        reject(err);
      }
      return resolve(stats.size);
    });
  });
}

function photoApiVideoUpload(args) {
  return Promise.resolve(getFileSize(args.filePath))
    .then(allsize => apiInit(args, allsize))
    .then((uploadInfo) => {
      return initUploadChain(args, args.filePath, uploadInfo);
    })
    .then((data) => {
      console.log('call apiFinish', data);
      debugger;
    });
}

module.exports = photoApiVideoUpload;
```

Example run:
```js
const path = require('path');
const GPhotos = require('upload-gphotos').default;
const photoUpload = require('photoApiVideoUpload'); //file above

const gphotos = new GPhotos({ username: 'my_email', password: 'my_password' });
  try {
    await gphotos.login();
    console.log('logged in');
    photoUpload({
      filePath: path.join(process.cwd(), 'SampleVideo_1280x720_1mb.mp4'),
      axios: gphotos.axios,
    });
    // const photo = await gphotos.upload();
    // console.log('upload', photo);
  } catch (error) {
    console.log('error', error);
  }
```

**Comment by tamdao** [comment]
@3846masa I think I need to create `MediaItem` with `upload id` after the upload completed. https://developers.google.com/photos/library/guides/upload-media#creating-media-item

**Comment by tamdao** [comment]
I have tried. But I don't see any request like `create media item` on photo.google.com So I tried with the api https://developers.google.com/photos/library/guides/upload-media#creating-media-item and it worked well.

@3846masa Do you have any idea?

**Comment by 3846masa** [comment]
Please check that document. It is a good document for understanding Google Photos.
https://github.com/simonedegiacomi/gphotosuploader/blob/8c01f4566c4937fb03c782a7b7ec16b467042071/documentation/how-upload-works.md

**Comment by tamdao** [comment]
Thanks for your info. I will create PR tonight

**Comment by animey** [comment]
Is this script bypass counter?
> All media items uploaded to Google Photos through the API are stored in full resolution at original quality. If your uploads exceed 25MB per user, your application should remind the user that these uploads will count towards storage in their Google Account.

**Comment by 3846masa** [comment]
It seems to be fixed by #334.
